install_requirements.py

```diff
@@ -0,0 +1,168 @@
+# LICENSE file in the root directory of this source tree.
+
+import os
+import platform
+import re
+import subprocess
+import sys
+
+# Before doing anything, cd to the directory containing this script.
+os.chdir(os.path.dirname(os.path.abspath(__file__)))
+
+
+def python_is_compatible():
+    # Scrape the version range from pyproject.toml, which should be in the current directory.
+    version_specifier = None
+    with open("pyproject.toml", "r") as file:
+        for line in file:
+            if line.startswith("requires-python"):
+                match = re.search(r'"([^"]*)"', line)
+                if match:
+                    version_specifier = match.group(1)
+                    break
+
+    if not version_specifier:
+        print(
+            "WARNING: Skipping python version check: version range not found",
+            file=sys.stderr,
+        )
+        return False
+
+    # Install the packaging module if necessary.
+    try:
+        import packaging
+    except ImportError:
+        subprocess.run(
+            [sys.executable, "-m", "pip", "install", "packaging"], check=True
+        )
+    # Compare the current python version to the range in version_specifier. Exits
+    # with status 1 if the version is not compatible, or with status 0 if the
+    # version is compatible or the logic itself fails.
+    try:
+        import packaging.specifiers
+        import packaging.version
+
+        python_version = packaging.version.parse(platform.python_version())
+        version_range = packaging.specifiers.SpecifierSet(version_specifier)
+        if python_version not in version_range:
+            print(
+                f'ERROR: ExecuTorch does not support python version {python_version}: must satisfy "{version_specifier}"',
+                file=sys.stderr,
+            )
+            return False
+    except Exception as e:
+        print(f"WARNING: Skipping python version check: {e}", file=sys.stderr)
+    return True
+
+
+if not python_is_compatible():
+    sys.exit(1)
+
+# Parse options.
+EXECUTORCH_BUILD_PYBIND = "OFF"
+CMAKE_ARGS = os.getenv("CMAKE_ARGS", "")
+CMAKE_BUILD_ARGS = os.getenv("CMAKE_BUILD_ARGS", "")
+
+for arg in sys.argv[1:]:
+    if arg == "--pybind":
+        EXECUTORCH_BUILD_PYBIND = "ON"
+    elif arg in ["coreml", "mps", "xnnpack"]:
+        if EXECUTORCH_BUILD_PYBIND == "ON":
+            arg_upper = arg.upper()
+            CMAKE_ARGS += f" -DEXECUTORCH_BUILD_{arg_upper}=ON"
+        else:
+            print(f"Error: {arg} must follow --pybind")
+            sys.exit(1)
+    else:
+        print(f"Error: Unknown option {arg}")
+        sys.exit(1)
+
+# Since ExecuTorch often uses main-branch features of pytorch, only the nightly
+# pip versions will have the required features.
+#
+# NOTE: If a newly-fetched version of the executorch repo changes the value of
+# NIGHTLY_VERSION, you should re-run this script to install the necessary
+# package versions.
+NIGHTLY_VERSION = "dev20240716"
```

Comment:
We're about to bump this in #4794. To make the merge easier, please update to the new version here
```suggestion
NIGHTLY_VERSION = "dev20240820"
```

Reply:
Oh I'm able to push to your branch. I made this fix.